### PR TITLE
DOC: special: small improvements to the `poch` docstring

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue

None.

#### What does this implement/fix?

This improves the docstring for `special.poch`.

#### Additional information

In particular,

- Add a reference
- Add examples

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -7097,17 +7097,47 @@ def add_newdoc(name, doc):
...
+    See [dlmf]_ for more details.
+
     Parameters
     ----------
-    z : array_like
-        (int or float)
-    m : array_like
-        (int or float)
+    z, m : array_like
+        Real-valued arguments.
```
Comment (tylerjereddy) [inline scipy/special/add_newdocs.py]: Isn't `array_like` not quite right for the type field since you can have a scalar input for either `z`  or `m`?

**Review by tylerjereddy — COMMENTED** [review]
Looks much nicer than the devdocs version when I do a side-by-side comparison of the rendered docstrings.

Just added one minor comment

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (rgommers) [reply scipy/special/add_newdocs.py]: `array_like` includes scalar.

**Comment by rgommers** [comment]
LGTM, merged. Thanks @person142

**Pull request merged**